### ISSUE TITLE
`smooth_add` doesn’t crossfade out, produces glitch at end of "special"

### ISSUE THREAD

**Issue by Moonbase59** [savonet/liquidsoap#3714]
`smooth_add` does crossfade in, but not out. And it makes a nasty glitch at the end of the "special" source.

Using LS 2.2.4+git@6721bb156 on Linux.

![test-smooth_add-1_001](https://github.com/savonet/liquidsoap/assets/3706922/96489007-2848-4b3c-a2d9-74d74bdb9020)

What I expected:
crossfade on _both_ ends of "special", with the duration specified. (Much like in LS 1.x.)

---

**What I _really_ want:** A smooth "top-of-the-hour" overlay on the radio stream. Maybe an option to specify whether the "special" source should start/stop _immediately_ (and thus being faded), or play only _within_ the faded-down "normal" source. This would make things easier "for the rest of us", by internally adding the required blanks.

I cannot find a way to simply add 2s of silence to both ends of each track taken from the `playlist_zeitansage`, which would achieve the same effect (i.e., fade "normal" down, play "special", fade normal up again). 

Desired outcome (but without the glitch at the end of "special"):

![Auswahl_333](https://github.com/savonet/liquidsoap/assets/3706922/22eb6be1-76c2-46ad-9467-ffa6d5d40019)

**Comment by rrolla** [comment]
Yeah I noticed the same when testing version `2.2.4, 2.2.5`, on my production version that's running on liquidsoap `v2.1.4` start and end fade is all good.

**Comment by ymatto** [comment]
> What I really want: A smooth "top-of-the-hour" overlay on the radio stream.

I found this thread wanting the same. I workaround that's working for me, If you're using a `request.queue`, is to create your `special` for the `smooth_add`, then surround your overlay with a pair of chunks of silence that you can then tune to suit.

`playover_source.push(request.create("synth:shape=sine,frequency=0.0,duration=3"))`
`playover_source.push(request.create("file:announcement.wav"))`
`playover_source.push(request.create("synth:shape=sine,frequency=0.0,duration=1"))`

Then the `duration` in your `smooth_add` has a little "padding" to work with.

Bit hacky, but it's working for me.
